File: WordTarget/WordTargetCore/LinkType.cs
using System;

namespace WordTargetCore
{
    public enum LinkType
    {
        OneLetterChange,
        OneLetterAddOrRemove,
        Anagram,
        Synonym,
        Antonym,
        WordAssociation
    }

    public static class LinkTypeExtensions
    {
        public static bool IsAutomatic(this LinkType type)
        {
            switch (type)
            {
                case LinkType.OneLetterChange:
                case LinkType.OneLetterAddOrRemove:
                case LinkType.Anagram:
                    return true;
                case LinkType.Synonym:
                case LinkType.Antonym:
                case LinkType.WordAssociation:
                default:
                    return false;
            }
        }
    }
}

File: word-target/word-target-core/src/main/java/it/carcassi/wordtarget/core/LinkType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package it.carcassi.wordtarget.core;

/**
 *
 * @author Matteo
 */
public enum LinkType {
    OneLetterChange,
    OneLetterAddOrRemove,
    Anagram,
    Synonym,
    Antonym,
    WordAssociation;

    public boolean isAutomatic() {
        switch (this) {
            case OneLetterChange:
            case OneLetterAddOrRemove:
            case Anagram:
                return true;
            case Synonym:
            case Antonym:
            case WordAssociation:
            default:
                return false;
        }
    }
}
